lib/mayu/vdom/hydration.rb

```diff
@@ -0,0 +1,78 @@
+# typed: strict
+
+module Mayu
+  module VDOM
+    module Hydration
```

Comment:
Rename this to `Marshalling` or something, hydration is something else.

What's happening in Mayu is closer to https://github.com/BuilderIO/qwik/blob/main/packages/docs/src/routes/docs/concepts/resumable/index.mdx